portfolio/portfolio.go

```diff
@@ -236,16 +237,14 @@ func (b *Base) RemoveAddress(address, description string, coinType currency.Code
 		return errors.New("coin type is empty")
 	}
 
-	for x := range b.Addresses {
-		if b.Addresses[x].Address == address &&
-			b.Addresses[x].CoinType.Equal(coinType) &&
-			b.Addresses[x].Description == description {
-			b.Addresses = append(b.Addresses[:x], b.Addresses[x+1:]...)
-			return nil
-		}
+	idx := slices.IndexFunc(b.Addresses, func(a Address) bool {
+		return a.Address == address && a.CoinType.Equal(coinType) && a.Description == description
+	})
+	if idx == -1 {
```

Comment:
🐒 Think this pattern is probably cleaner and more efficient, and only mentioning it so we don't cargo-cult whichever we don't choose.
```suggestion
	oldSize := len(b.Addresses)
	b.Addresses = slices.Clip(slices.DeleteFunc(b.Addresses, func(a Address) bool {
		return a.Address == address && a.CoinType.Equal(coinType) && a.Description == description
	}))
	if len(b.Addresses) == oldSize {
```